freqtrade/freqai/data_drawer.py

```diff
@@ -293,6 +294,9 @@ def append_model_predictions(self, pair: str, predictions: DataFrame,
             for return_str in rets:
                 df[return_str].iloc[-1] = rets[return_str]
 
+        df['close_price'].iloc[-1] = strat_df['close'].iloc[-1]
+        df['date_pred'].iloc[-1] = strat_df['date'].iloc[-1]
```

Comment:
shouldn't close and date already be part of the dataframe when it comes in? 

will this be a breaking change - requiring identifier resets?

I'm also not sure on the "post processing options".
if i look at the below - then `.df` has a rolling length of the incoming dataframe.
That's for sure correct, but it'll mean that in the end, the data will age out.

Reply:
Historic predictions dataframe is not the same as the incoming strategy dataframe.

if users update versions they need to ensure a fresh identifier.

Reply:
Historic predictions is an indefinite record of the bots run - you don’t phase historic predictions out. This is data for post processing.